telemetry/statsd/src/main/scala/io/buoyant/telemetry/statsd/StatsDStatsReceiver.scala

```diff
@@ -1,25 +1,24 @@
 package io.buoyant.telemetry.statsd
 
 import java.util.concurrent.ConcurrentHashMap
+
 import com.timgroup.statsd.StatsDClient
-import com.twitter.finagle.stats.{Counter, Stat, StatsReceiverWithCumulativeGauges, Verbosity}
-import scala.collection.JavaConverters._
-
-private[telemetry] object StatsDStatsReceiver {
-  // from https://github.com/researchgate/diamond-linkerd-collector/
-  private[statsd] def mkName(name: Seq[String]): String = {
-    name.mkString("/")
-      .replaceAll("[^/A-Za-z0-9]", "_")
-      .replace("//", "/")
-      .replace("/", ".") // https://graphite.readthedocs.io/en/latest/feeding-carbon.html#step-1-plan-a-naming-hierarchy
-  }
+import com.twitter.finagle.stats.{
+  Counter,
+  CounterSchema,
+  HistogramSchema,
```

Comment:
ah, I see where the escaping logic came from now.  this escaping should be restricted to the statsd stats receiver and not be used by the metrics tree stats receiver.

Reply:
I made the assumption that the names for StatsDStatsReceiver and MetricsTreeStatsReceiver were in the same format and tried to consolidate some code.

In the process of reverting that change now.

To implement `counter` and `stat` in the `MetricsTreeStatsReceiver`, I've got this code to call the overridden methods:

```scala
  def counter(schema: CounterSchema): Counter =
    counter(verbosity, schema.metricBuilder.name.head)

  def stat(schema: HistogramSchema): Stat =
    stat(verbosity, schema.metricBuilder.name.head)
```

Since `MetricsTree.resolve()` [checks for Nil](https://github.com/linkerd/linkerd/blob/master/telemetry/core/src/main/scala/io/buoyant/telemetry/MetricsTree.scala#L55), I don't think I have to worry about checking `schema.metricBuilder.name.head` here. Does that sound right?

Reply:
I think this should be `counter(verbosity, schema.metricsBuilder.name: _*)` so that all the elements of the name seq get passes as varargs.

Reply:
Thanks, that syntax is new to me, and it makes sense.

Will commit with the change shortly.